vendor/fasttext/ffi_fasttext.cc

```diff
@@ -22,7 +23,18 @@
 EXTERN_C_BEGIN
 fasttext::FastText* create(const char* model_name) {
   fasttext::FastText* new_fasttext = new fasttext::FastText();
-  new_fasttext->loadModel(std::string(model_name));
+  std::string model{model_name};
+  try {
+    if (model.substr(0, 4) == "http") {
```

Comment:
this mechanism to load is just too brittle, aren't there other ways we could achieve the same outcome? (a rake task that download from remote server?) ... seems like production loading from remote is also not a good idea, this is something that I would put in a runtime hook in ruby and download the remote source

Reply:
The download isn't really remote, it's from our own internal ceph server. We've explored putting the model files in git using git-lfs, but there are some drawbacks of that method that I think Matt could explain better. Disk space is one issue, I believe.

Reply:
yeah, the problem is that this is an open source gem and not a private one ... I think we should create a better interface, we can add an additional method that loads from an http source instead of branching on 'http', creates a brittle interface